### PR TITLE
python3Packages.pynx584: init at 0.6

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python package for communicating to NX584/NX8E interfaces

https://github.com/kk7ds/pynx584

This is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 120272` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.pynx584</li>
    <li>python39Packages.pynx584</li>
  </ul>
</details>

File: pkgs/development/python-modules/pynx584/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  propagatedBuildInputs = [
+    flask
+    prettytable
+    pyserial
+    requests
+    stevedore
+  ];
+
+  # No GitHub releases available, tests are not present in PyPI release
```
Comment (mweinelt) [inline pkgs/development/python-modules/pynx584/default.nix:30]: https://github.com/kk7ds/pynx584/issues/55

It always helps to make upstream aware of our needs. That being said, can you fetch from GitHub and enable tests and let's hope we get the tag on the next update.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120272` at ca73a19d run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.pynx584</li>
    <li>python39Packages.pynx584</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
diff LGTM

**Review by mweinelt — CHANGES_REQUESTED** [review]
Please switch to Git, even if there is no tag, and enable tests.

File: pkgs/development/python-modules/pynx584/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pynx584/default.nix:30]: Tag is up.

**Pull request merged**